src/pages/signin/SignInPage.js

```diff
@@ -0,0 +1,158 @@
+    session: PropTypes.shape({
+        // Error to display when there is a session error returned
+        error: PropTypes.string,
+    }),
+};
+
+const defaultProps = {
+    account: {},
+    session: null,
+    credentials: {},
+};
+
+class SignInPage extends Component {
+    componentDidMount() {
+        // Always reset the unread counter to zero on this page
+        updateUnread(0);
+    }
+
+    render() {
+        const session = this.props.session || {};
+
+        // If we end up on the sign in page and have an authToken then
+        // we are signed in and should be brought back to the site root
+        if (session.authToken) {
+            return <Redirect to={ROUTES.ROOT} />;
+        }
+
+        // @TODO figure out the real logic that needs to be used for showing the proper form
+        // It will depend on if they had an account already, had a Github username already,
+        // if they have access to this application, if they need to validate their account, etc.
+
+        // Show the login form if
+        // - A login has not been entered yet
+        const showLoginForm = !this.props.credentials.login;
+
+        // Show the GitHub username form if
+        // - A login has been entered
+        // - AND they do not have access to this app yet
+        // - AND the user hasn't entered a GitHub username yet
+        // - AND a password hasn't been entered yet
+        const showGithubUsernameForm = this.props.credentials.login
+            && !this.props.account.canAccessExpensifyCash
+            && !this.props.credentials.githubUsername
+            && !this.props.credentials.password;
+
+        // Show the password form if
+        // - A login has been entered
+        // - AND a GitHub username has been entered OR they already have access to expensify cash
+        // - AND an account exists already for this login
+        // - AND a password hasn't been entered yet
+        const showPasswordForm = this.props.credentials.login
+            && (
+                this.props.credentials.githubUsername
+                || this.props.account.canAccessExpensifyCash
+            )
+            && this.props.account.accountExists
+            && !this.props.credentials.password;
+
+        // Show the resend validation link form if
+        // - A login has been entered
+        // - AND a GitHub username has been entered OR they already have access to this app
+        // - AND an account did not exist for that login
+        const showResendValidationLinkForm = this.props.credentials.login
+            && (
+                this.props.credentials.githubUsername
+                || this.props.account.canAccessExpensifyCash
+            )
+            && !this.props.account.accountExists;
+
+        return (
+            <>
+                <CustomStatusBar />
+                <SafeAreaView style={[styles.signInPage]}>
+                    <View style={[styles.signInPageInner]} accessibilityRole="form">
```

Comment:
I don't think iOS likes this `accessibilityRole`:

<img width="497" alt="Screen Shot 2020-12-21 at 11 35 29 AM" src="https://user-images.githubusercontent.com/2838819/102815110-b23ab980-4380-11eb-8761-d0364d44fc6f.png">

Reply:
Thanks, Andrew. I was going to look at mobile once all the APIs were working, but you beat me to it! I'll just remove it for now since it only results in a browser warning. Sounds like we'll need to do a no-op for this on mobile.